bitnami/mongodb-sharded/templates/_helpers.tpl

```diff
@@ -0,0 +1,375 @@
+Get the initialization scripts Secret name.
+*/}}
+{{- define "mongodb-sharded.initScriptsSecret" -}}
+  {{- printf "%s" (include "mongodb-sharded.tplValue" (dict "value" .Values.initScriptsSecret "context" $)) -}}
+{{- end -}}
+
+{{/*
+Get the initialization scripts configmap name.
+*/}}
+{{- define "mongodb-sharded.initScriptsCM" -}}
+  {{- printf "%s" (include "mongodb-sharded.tplValue" (dict "value" .Values.initScriptsCM "context" $)) -}}
+{{- end -}}
+
+{{/*
+Create the name for the admin secret.
+*/}}
+{{- define "mongodb-sharded.adminSecret" -}}
+    {{- if .Values.auth.existingAdminSecret -}}
+        {{- .Values.auth.existingAdminSecret -}}
+    {{- else -}}
+        {{- include "mongodb-sharded.fullname" . -}}-admin
+    {{- end -}}
+{{- end -}}
+
+{{/*
+Create the name for the key secret.
+*/}}
+{{- define "mongodb-sharded.keySecret" -}}
+  {{- if .Values.auth.existingKeySecret -}}
+      {{- .Values.auth.existingKeySecret -}}
+  {{- else -}}
+      {{- include "mongodb-sharded.fullname" . -}}-keyfile
+  {{- end -}}
+{{- end -}}
+
+{{/*
+Return the proper MongoDB image name
+*/}}
+{{- define "mongodb-sharded.image" -}}
+  {{- $registryName := .Values.image.registry -}}
+  {{- $repositoryName := .Values.image.repository -}}
+  {{- $tag := .Values.image.tag | toString -}}
+  {{/*
+  Helm 2.11 supports the assignment of a value to a variable defined in a different scope,
+  but Helm 2.9 and 2.10 doesn't support it, so we need to implement this if-else logic.
+  Also, we can't use a single if because lazy evaluation is not an option
+  */}}
+  {{- if .Values.global }}
+    {{- if .Values.global.imageRegistry }}
+      {{- printf "%s/%s:%s" .Values.global.imageRegistry $repositoryName $tag -}}
+    {{- else -}}
+      {{- printf "%s/%s:%s" $registryName $repositoryName $tag -}}
+    {{- end -}}
+  {{- else -}}
+    {{- printf "%s/%s:%s" $registryName $repositoryName $tag -}}
+  {{- end -}}
+{{- end -}}
+
+{{/*
+Return the proper image name (for the metrics image)
+*/}}
+{{- define "mongodb-sharded.metrics.image" -}}
+  {{- $registryName := .Values.metrics.image.registry -}}
+  {{- $repositoryName := .Values.metrics.image.repository -}}
+  {{- $tag := .Values.metrics.image.tag | toString -}}
+  {{/*
+  Helm 2.11 supports the assignment of a value to a variable defined in a different scope,
+  but Helm 2.9 and 2.10 doesn't support it, so we need to implement this if-else logic.
+  Also, we can't use a single if because lazy evaluation is not an option
+  */}}
+  {{- if .Values.global }}
+    {{- if .Values.global.imageRegistry }}
+      {{- printf "%s/%s:%s" .Values.global.imageRegistry $repositoryName $tag -}}
+    {{- else -}}
+      {{- printf "%s/%s:%s" $registryName $repositoryName $tag -}}
+    {{- end -}}
+  {{- else -}}
+    {{- printf "%s/%s:%s" $registryName $repositoryName $tag -}}
+  {{- end -}}
+{{- end -}}
+
+
+{{/*
+Return the proper Docker Image Registry Secret Names
+*/}}
+{{- define "mongodb-sharded.imagePullSecrets" -}}
```

Comment:
There is a simpler way of defining these imagePullSecrets helper templates. Check minio for example and the use of the coalesce function.

https://github.com/bitnami/charts/blob/master/bitnami/minio/templates/_helpers.tpl#L54-L65